### PR TITLE
[feat] Fatal vs. Non-fatal error attribution

### LINKED ISSUE #155: ROM flow error attribution review
Classify errors into fatal vs non-blocking

### PR DESCRIPTION
**Pull request by mhatrevi**
This change documents the fatal (cold reset) and non-fatal errors (update-reset) raised by the ROM. Additionally, it fixes few error reporting discrepancies.

### CODE REVIEW

File: rom/dev/doc/error-attribution.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
+# **Fatal Errors**
```
Comment (korran) [inline rom/dev/doc/error-attribution.md:1]: Is there any information here that's not already in `error/src/lib.rs`? Manually maintaining a duplicate seems like a waste of time to me...

File: hw-model/src/lib.rs [hunk 2]
```diff
@@ -569,7 +571,11 @@ pub trait HwModel {
             self.soc_mbox().execute().write(|w| w.execute(false));
             let soc_ifc = self.soc_ifc();
             return Err(ModelError::MailboxCmdFailed(
-                soc_ifc.cptra_fw_error_non_fatal().read(),
+                if cmd == MBOX_DOWNLOAD_FIRMWARE_CMD_ID {
```
Comment (korran) [inline hw-model/src/lib.rs:574]: Eh, I don't really like this sort of hard-coded logic. Can we put the error in both fw_error_fatal and fw_error_non_fatal?

File: rom/dev/doc/error-attribution.md (hunk 1 above)
Comment (mhatrevi) [reply rom/dev/doc/error-attribution.md:1]: Errors in lib.rs don't have an attribution of fatal vs. non-fatal. Eventually, this document can get published as official.

File: hw-model/src/lib.rs (hunk 2 above)
Comment (mhatrevi) [reply hw-model/src/lib.rs]: Discussed offline; filed https://github.com/chipsalliance/caliptra-sw/issues/453 to add fatal error to non-fatal registers for keeping test functions generic.

File: rom/dev/doc/error-attribution.md (hunk 1 above)
Comment (mhatrevi) [reply rom/dev/doc/error-attribution.md:1]: Discussed offline; we'll keep this for now and auto-generate it later. At that time, this file should be removed.

**Review by korran — APPROVED** [review]

**Pull request merged**